src/CSModule.sol

```diff
@@ -964,27 +1024,48 @@ contract CSModule is
                 }
             }
         }
-        if (loadedKeysCount != depositsCount) {
-            revert NotEnoughKeys();
-        }
 
-        unchecked {
-            // @dev depositsCount can not overflow in practice due to memory and gas limits
-            _depositableValidatorsCount -= uint64(depositsCount);
-            _totalDepositedValidators += uint64(depositsCount);
+        return (depositsLeft, loadedKeysCount);
+    }
+
+    // TODO: Remove with fixing tests.
+    function depositQueue() external view returns (uint128 head, uint128 tail) {
+        return (legacyQueue.head, legacyQueue.tail);
+    }
+
+    function registerDepositPriority(
+        uint256 value
+    ) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        // Ensured by an upgrade to v2.
+        // if (priority == LEGACY_QUEUE_PRIOIRITY) {
+        //     revert QueuePriorityAlreadyUsed();
+        // }
+
+        for (uint256 i; i < _depositPriorities.length; ++i) {
+            uint256 buf = _depositPriorities[i];
+
+            if (value == buf) {
+                revert DepositPriorityAlreadyUsed();
+            }
+
+            if (value < buf) {
+                _depositPriorities[i] = value;
+                value = buf;
+            }
         }
 
-        _incrementModuleNonce();
+        _depositPriorities.push(value);
     }
 
     /// @inheritdoc ICSModule
     function cleanDepositQueue(
         uint256 maxItems
     ) external returns (uint256 removed, uint256 lastRemovedAtDepth) {
-        (removed, lastRemovedAtDepth) = depositQueue.clean(
-            _nodeOperators,
-            maxItems
-        );
+        for (uint256 p; p < _depositPriorities.length; ++p) {
```

Comment:
What about adding a queuePriority as a parameter here and cleaning only one queue at a time? Might be more complicated if multiple queues are polluted, but this is a rare case

Reply:
A few considerations I have:
1) We won't be able to discern whether a batch is valid if a node operator has entries in multiple queues, which is priority+normal as an example.
2) We potentially kick the guys out of the queue even though they can fix their situation (which is currently is better handled by cleaning from the beginning only).